### PR TITLE
CFINSPEC-66 : Add Ipfilter resource

### PR DESCRIPTION
**Pull request by ahasunos**
Signed-off-by: Sonu Saha <sonu.saha@progress.com>

## Description

ISSUE: CFINSPEC-66 — Create ipfilter resource. `ipfilter` resource helps to test rules that are defined for ipfilter (also known as IPF) firewall.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Changes made to existing file:
* test/helpers/mock_loader.rb

## New files included for ipfilter resource:
* docs-chef-io/content/inspec/resources/ipfilter.md
* lib/inspec/resources/ipfilter.rb
* test/fixtures/cmd/ipfstat-io
* test/unit/resources/ipfilter_test.rb

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Review by clintoncwolfe — APPROVED** [review]
This looks great! Good work!

**Comment by clintoncwolfe** [comment]
Sigh. all the test failures can be ignored for various bad reasons.

**Review by Vasu1105 — APPROVED** [review]
LGTM too. Good Work Sonu !!

**Comment by Nik08** [comment]
If only three people could approve this code, I would have done it too :D 
Looks good and super clean too <3

**Comment by ahasunos** [comment]
Thank you team 🙌  @clintoncwolfe, @Vasu1105, @Nik08

File: docs-chef-io/content/inspec/resources/ipfilter.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
++++
+title = "ipfilter resource"
+draft = false
+gh_repo = "inspec"
+platform = "bsd"
+platform = "solaris"
```
Comment (IanMadd) [inline docs-chef-io/content/inspec/resources/ipfilter.md]: A couple issues here:

1. This platform parameter is used to sort resource pages in this page: https://docs.chef.io/inspec/resources/ and we don't currently have bsd or solaris platforms listed there (although we could add them)

2. The toml spec doesn't accept repeating entries. 

It's possible that we could define this as a list of platforms, but I don't think that will work on the resource list page right now.

So I guess I have two questions: 
1. Would this page fit under one of the existing headings on the [inspec/resources](https://docs.chef.io/inspec/resources/) page? (I'm guessing not)
2. Will there be a bunch more resources that are in bsd and/or solaris? In which case I can add headings for both and figure out how to handle these platforms in a list.

Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/ipfilter.md]: @ahasunos - please clarify the questions raised by Ian.

Comment (clintoncwolfe) [reply docs-chef-io/content/inspec/resources/ipfilter.md]: There will be several more resources that are bsd/ solaris only. As bsd is more popular than solaris, if we're only allowed one tag, I'd go with bsd.

**Pull request merged**